scripts/calculateRevenue/protocols/somm/dailySnapshots.ts

```diff
@@ -0,0 +1,95 @@
+import fetch from 'node-fetch'
+import { DailySnapshot } from './types'
+import { NetworkId } from '../../../types'
+
+const TWENTY_FOUR_HOURS = 24 * 60 * 60 * 1000
+
+const NETWORK_ID_TO_CHAIN_ID: Partial<Record<NetworkId, string>> = {
+  [NetworkId['ethereum-mainnet']]: 'ethereum',
+  [NetworkId['arbitrum-one']]: 'arbitrum',
+  [NetworkId['op-mainnet']]: 'optimism',
+  [NetworkId['polygon-pos-mainnet']]: 'polygon',
+  [NetworkId['base-mainnet']]: 'base',
+}
+
+/**
+ * https://api.sommelier.finance/info
+ */
+export async function getDailySnapshots({
+  networkId,
+  vaultAddress,
+  startTimestamp,
+  endTimestamp,
+}: {
+  networkId: NetworkId
+  vaultAddress: string
+  startTimestamp: Date
+  endTimestamp: Date
+}): Promise<DailySnapshot[]> {
+  // Subtract 24 hours from the start time to ensure we get the snapshot for the start time
+  const startUnixTimestamp = Math.floor(
+    (startTimestamp.getTime() - TWENTY_FOUR_HOURS) / 1000,
+  )
+  const endUnixTimestamp = Math.floor(endTimestamp.getTime() / 1000)
+  const url = `https://api.sommelier.finance/dailyData/${NETWORK_ID_TO_CHAIN_ID[networkId]}/${vaultAddress}/${startUnixTimestamp}/${endUnixTimestamp}`
```

Comment:
Are the returned snapshots here guaranteed to be sorted by date?

Reply:
They are not, but my usd price calculator doesn't require them to be sorted